src/app/src/main/java/com/example/bookmark/adapters/BookList.java

```diff
@@ -68,15 +71,25 @@ public View getView(int position, @Nullable View convertView, @NonNull ViewGroup
         String bookStatus = ("Status: "
             + book.getStatus().toString().charAt(0)
             + book.getStatus().toString().substring(1).toLowerCase());
-        // TODO get owner name once functionality is supported by Owner class
-        String bookOwner = "Owner: "; // + book.getOwner().toString();
+        String bookOwner = "Owner: " + book.getOwnerId().toString();
 
         title.setText(book.getTitle());
         author.setText(book.getAuthor());
         description.setText(book.getDescription());
         owner.setText(bookOwner);
         status.setText(bookStatus);
 
+        switch(this.activityTitle) {
+            case "My Books":
+                if (book.getStatus().toString().equals("REQUESTED")) {
+                    notificationIcon.setVisibility(View.VISIBLE);
+                } break;
+            case "Pending Requests":
+                if (book.getStatus().toString().equals("ACCEPTED")) {
+                    notificationIcon.setVisibility(view.VISIBLE);
+                } break;
+        }
+
```

Comment:
The formatting here is a little strange, but more importantly, should this have a default case for when the string is neither of the listed cases?

Reply:
Actually, it might be better to pass in the Activity's class. That way you don't hardcode the same strings in two places (if we renamed a page this would no longer work and be painful to debug).

Reply:
The visibility is set to hidden by default in the uml? Do you think it is better to move setting visibility to the switch case? I''ll fix the formatting

Reply:
yeah I agree